bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/SubMonitor.java

```diff
@@ -784,6 +785,38 @@ public SubMonitor newChild(int totalWork) {
 		return newChild(totalWork, SUPPRESS_BEGINTASK);
 	}
 
+	@Override
+	public IProgressMonitor create(int totalWork, ProgressOption... options) {
+		if (options.length == 0) {
+			return newChild(totalWork);
+		}
+		if (options.length == 1 && options[0] == ProgressOption.SUPPRESS_SUBTASK) {
+			return newChild(totalWork, SUPPRESS_BEGINTASK | SUPPRESS_SUBTASK);
+		}
+		// use default implementation for ProgressOption.PREPEND_MAIN_LABEL:
+		return create(this, totalWork, options);
+	}
+
+	static IProgressMonitor create(IProgressMonitor mon, int totalWork, ProgressOption[] options) {
+		if (options.length == 0) {
```

Comment:
As a very first step one should check here if the passed in IProgressMonitor is already an instance of submonitor and then transform this into a split.

Reply:
thats not needed: In the case of Submonitor the overload org.eclipse.core.runtime.SubMonitor.submonitor(int, ProgressOption...) has already checked that.

Reply:
Okay its a bit confusing can this method maybe moved to "Sub**Progress**Monitor" or even just add this impl to the default of `IProgressMonitor` to make it more clear that this is actually nothing that belongs to `SubMonitor`?

Reply:
> default of `IProgressMonitor` to make it more clear that this is actually nothing that belongs to `SubMonitor`?

I started with that, but its a bit confusing to call the default implementation fallback from SubMonitor since its super interface is deprecated. As soon as IProgressMonitorWithBlocking is removed that would be the better way.

Reply:
> I started with that, but its a bit confusing to call the default implementation fallback from SubMonitor since its super interface is deprecated. 

As mentioned before, `SubMonitor` should never need to call any default implementation it should fully implement the API itself so we can prove the new API is actually implementable (without rely on old deprecated API).